File: 0014-longest-common-prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = "";

        for(int i=0; i<strs[0].length(); i++){
            for(int j = 1; j<strs.size(); j++){
                if(strs[j].length() >= i+1){
                   if(strs[0][i] != strs[j][i]) return prefix;
                }else{
                    return prefix;
                }
            }
            prefix += strs[0][i];
        }
        return prefix;
    }
};

File: 0014-longest-common-prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let ansPrefix = "";
    for(let i=0; i<strs[0].length; i++){
        let ch = strs[0][i];
        let flag = 0;
        for(let j=1; j<strs.length; j++){
            if(strs[j].length < i+1){
                flag = 1;
                break ;
            }else{
                if(strs[j][i]  != ch){
                    flag = 1;
                    break;
                }
            }
        }
        if(flag == 1) return ansPrefix;
        ansPrefix+=ch;
    }

    return ansPrefix;
};
